src/index.js

```diff
@@ -387,8 +369,17 @@ Penpal.connectToParent = ({ parentOrigin, methods = {} }) => {
       child.removeEventListener(MESSAGE, handleMessageEvent);
       reject('Child: Connection destroyed');
     })
-  });
 
+    const sendHandshakeMessage = function() {
+      parent.postMessage({
+	    penpal: HANDSHAKE,
+	    methodNames: Object.keys(methods),
+      }, document.referrer);
```

Comment:
I don't think `document.referrer` will always always be a valid origin because it contains the file path as well. We'll need to pass `document.referrer` into `getOriginFromUrl` and use the result.